host-spec/c06-consensus.tm

```diff
@@ -220,11 +275,11 @@
     <math|\<cal-E\><rsub|n>> by <math|sc<rsub|n>>. <math|sc<rsub|n>> is set
     to the <verbatim|duration> field in the returned data from the call of
     the Runtime entry <verbatim|BabeApi_configuration> (see
-    <reference|sect-rte-babeapi-epoch>) at the beginning of each epoch. For a
-    given block <math|B>, we use the notation <strong|<math|s<rsub|B>>> to
-    refer to the slot during which <math|B> has been produced. Conversely,
-    for slot <math|s>, <math|\<cal-B\><rsub|s>> is the set of Blocks
-    generated at slot <math|s>.
+    <reference|sect-rte-babeapi-epoch>) at genesis. For a given block
```

Comment:
Why is the parameter `sc` obtained from the genesis? Instead of obtaining  `sc` from the genesis block, obtaining it from e.g. the first block of an epoch makes the parameter update more flexible. For example, if we change the parameter `c` later, we have to change `sc` as well.

Reply:
I choose this for the same reason epoch length is fixed at genesis. Substrate currently can not easily handle interval changes. It would be worth while discussing this with the implementers.